internal/manifests/collector/configmap_test.go

```diff
@@ -36,3 +49,228 @@ func TestDesiredConfigMap(t *testing.T) {
 
 	})
 }
+
+func TestDesiredPrometheusConfigMap(t *testing.T) {
```

Comment:
Could we add a test case where target allocator is disabled

Reply:
It's implicitly disabled in the first test case, is that okay or do you want it explicitly disabled?